internal/drand-cli/dkg_cli.go

```diff
@@ -766,6 +736,29 @@ func generateProposalCmd(c *cli.Context, l log.Logger) error {
 		proposalFile.Joining = append(proposalFile.Joining, p)
 	}
 
+	// for migration from v1-v2, we need to actually get the updated public keys of the remainers and leavers
+	for _, remainer := range remainers {
+		p, err := fetchPublicKey(beaconID, l, remainer, sch)
+		if err != nil {
+			return err
+		}
+		proposalFile.Remaining = append(proposalFile.Remaining, p)
+	}
+
+	for _, leaver := range leavers {
```

Comment:
Why do we need the leaver's pubkeys?

Reply:
we don't _really_, it's just a best effort. The reason it's here is because we check that leavers were remainers in the past, and because we check by `drand.Participant` and not just `drand.Participant.Address`, we need to migrate their keys over too (if possible)